ICE on neofs contract

<!-- Provide a general summary of the issue in the Title above -->

## Current Behavior
```
/home/rik/dev/neofs-contract/bin/cli contract compile -i contracts/neofs -c contracts/neofs/config.yml -m contracts/neofs/manifest.json -o contracts/neofs/contract.nef --bindings contracts/neofs/bindings_config.yml
panic: runtime error: invalid memory address or nil pointer dereference
        panic: runtime error: slice bounds out of range [:-1]
        panic: runtime error: slice bounds out of range [:-1]
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0xfd9cfe]

goroutine 1 [running]:
github.com/nspcc-dev/neo-go/pkg/compiler.(*varScope).dropScope(...)
        github.com/nspcc-dev/neo-go@v0.121.0/pkg/compiler/vars.go:42
panic({0x1311e00?, 0x1d381ac85008?})
        runtime/panic.go:860 +0x13a
github.com/nspcc-dev/neo-go/pkg/compiler.(*varScope).dropScope(...)
        github.com/nspcc-dev/neo-go@v0.121.0/pkg/compiler/vars.go:42
panic({0x11fd940?, 0x2052980?})
        runtime/panic.go:860 +0x13a
github.com/nspcc-dev/neo-go/pkg/compiler.(*codegen).Visit(0x1d381ae145a0, {0x1508298?, 0x1d381ae92cf0})
        github.com/nspcc-dev/neo-go@v0.121.0/pkg/compiler/codegen.go:1540 +0x7a9e
go/ast.Walk({0x1501500?, 0x1d381ae145a0?}, {0x1508298, 0x1d381ae92cf0})
        go/ast/walk.go:34 +0x4c
github.com/nspcc-dev/neo-go/pkg/compiler.(*codegen).emitLoadVar(0x1d381ae145a0, {0x0?, 0x1d381ae976a0?}, {0x153e510?, 0x0?})
        github.com/nspcc-dev/neo-go@v0.121.0/pkg/compiler/codegen.go:323 +0x2cc
github.com/nspcc-dev/neo-go/pkg/compiler.(*codegen).Visit(0x1d381ae145a0, {0x15082e8?, 0x1d381ae976a0})
        github.com/nspcc-dev/neo-go@v0.121.0/pkg/compiler/codegen.go:1032 +0x33f
go/ast.Walk({0x1501500?, 0x1d381ae145a0?}, {0x15082e8, 0x1d381ae976a0})
        go/ast/walk.go:34 +0x4c
github.com/nspcc-dev/neo-go/pkg/compiler.(*codegen).convertSyscall(0x1d381ae145a0, 0x1d381a9502c0, 0x1d381afa03c0)
        github.com/nspcc-dev/neo-go@v0.121.0/pkg/compiler/codegen.go:1886 +0x12c
github.com/nspcc-dev/neo-go/pkg/compiler.(*codegen).Visit(0x1d381ae145a0, {0x15082c0?, 0x1d381afa03c0})
        github.com/nspcc-dev/neo-go@v0.121.0/pkg/compiler/codegen.go:1206 +0x594c
go/ast.Walk({0x1501500?, 0x1d381ae145a0?}, {0x15082c0, 0x1d381afa03c0})
        go/ast/walk.go:34 +0x4c
github.com/nspcc-dev/neo-go/pkg/compiler.(*codegen).Visit-range4(...)
        github.com/nspcc-dev/neo-go@v0.121.0/pkg/compiler/codegen.go:873
github.com/nspcc-dev/neo-go/pkg/compiler.(*codegen).Visit.Backward[...].func3(...)
        slices/iter.go:29
github.com/nspcc-dev/neo-go/pkg/compiler.(*codegen).Visit(0x1d381ae145a0, {0x1508450?, 0x1d381ae976c0})
        github.com/nspcc-dev/neo-go@v0.121.0/pkg/compiler/codegen.go:871 +0x8a1d
go/ast.Walk({0x1501500?, 0x1d381ae145a0?}, {0x1508450, 0x1d381ae976c0})
        go/ast/walk.go:34 +0x4c
github.com/nspcc-dev/neo-go/pkg/compiler.(*codegen).Visit(0x1d381ae145a0, {0x1508388?, 0x1d381afa61b0})
        github.com/nspcc-dev/neo-go@v0.121.0/pkg/compiler/codegen.go:1361 +0x3f9c
go/ast.Walk({0x1501500?, 0x1d381ae145a0?}, {0x1508388, 0x1d381afa61b0})
        go/ast/walk.go:34 +0x4c
github.com/nspcc-dev/neo-go/pkg/compiler.(*codegen).inlineCall(0x1d381ae145a0, 0x1d381ab4abb0, 0x1d381ab31a00)
        github.com/nspcc-dev/neo-go@v0.121.0/pkg/compiler/inline.go:124 +0xe7d
github.com/nspcc-dev/neo-go/pkg/compiler.(*codegen).Visit(0x1d381ae145a0, {0x15082c0?, 0x1d381ab31a00})
        github.com/nspcc-dev/neo-go@v0.121.0/pkg/compiler/codegen.go:1119 +0x1b29
go/ast.Walk({0x1501500?, 0x1d381ae145a0?}, {0x15082c0, 0x1d381ab31a00})
        go/ast/walk.go:34 +0x4c
github.com/nspcc-dev/neo-go/pkg/compiler.(*codegen).Visit(0x1d381ae145a0, {0x1508298?, 0x1d381ae92d50})
        github.com/nspcc-dev/neo-go@v0.121.0/pkg/compiler/codegen.go:1534 +0x7a4d
go/ast.Walk({0x1501500?, 0x1d381ae145a0?}, {0x1508298, 0x1d381ae92d50})
        go/ast/walk.go:34 +0x4c
github.com/nspcc-dev/neo-go/pkg/compiler.(*codegen).Visit-range4(...)
        github.com/nspcc-dev/neo-go@v0.121.0/pkg/compiler/codegen.go:873
github.com/nspcc-dev/neo-go/pkg/compiler.(*codegen).Visit.Backward[...].func3(...)
        slices/iter.go:29
github.com/nspcc-dev/neo-go/pkg/compiler.(*codegen).Visit(0x1d381ae145a0, {0x1508450?, 0x1d381abddde0})
        github.com/nspcc-dev/neo-go@v0.121.0/pkg/compiler/codegen.go:871 +0x8a1d
go/ast.Walk({0x1501500?, 0x1d381ae145a0?}, {0x1508450, 0x1d381abddde0})
        go/ast/walk.go:34 +0x4c
github.com/nspcc-dev/neo-go/pkg/compiler.(*codegen).Visit(0x1d381ae145a0, {0x1508388?, 0x1d381ae92d80})
        github.com/nspcc-dev/neo-go@v0.121.0/pkg/compiler/codegen.go:1361 +0x3f9c
go/ast.Walk({0x1501500?, 0x1d381ae145a0?}, {0x1508388, 0x1d381ae92d80})
        go/ast/walk.go:34 +0x4c
github.com/nspcc-dev/neo-go/pkg/compiler.(*codegen).Visit(0x1d381ae145a0, {0x1508478?, 0x1d381ab31a40})
        github.com/nspcc-dev/neo-go@v0.121.0/pkg/compiler/codegen.go:905 +0x4a25
go/ast.Walk({0x1501500?, 0x1d381ae145a0?}, {0x1508478, 0x1d381ab31a40})
        go/ast/walk.go:34 +0x4c
github.com/nspcc-dev/neo-go/pkg/compiler.(*codegen).Visit(0x1d381ae145a0, {0x1508388?, 0x1d381ae93500})
        github.com/nspcc-dev/neo-go@v0.121.0/pkg/compiler/codegen.go:1361 +0x3f9c
go/ast.Walk({0x1501500?, 0x1d381ae145a0?}, {0x1508388, 0x1d381ae93500})
        go/ast/walk.go:34 +0x4c
github.com/nspcc-dev/neo-go/pkg/compiler.(*codegen).convertFuncDecl(0x1d381ae145a0, {0x15083d8, 0x1d381a9c01e0}, 0x1d381ae93950, 0x1d381aa25800)
        github.com/nspcc-dev/neo-go@v0.121.0/pkg/compiler/codegen.go:610 +0x8dc
github.com/nspcc-dev/neo-go/pkg/compiler.(*codegen).compile.func2(...)
        github.com/nspcc-dev/neo-go@v0.121.0/pkg/compiler/codegen.go:2509
github.com/nspcc-dev/neo-go/pkg/compiler.(*codegen).compile.(*codegen).ForEachFile.func4(...)
        github.com/nspcc-dev/neo-go@v0.121.0/pkg/compiler/compiler.go:133
github.com/nspcc-dev/neo-go/pkg/compiler.(*codegen).ForEachPackage(...)
        github.com/nspcc-dev/neo-go@v0.121.0/pkg/compiler/compiler.go:124
github.com/nspcc-dev/neo-go/pkg/compiler.(*codegen).ForEachFile(...)
        github.com/nspcc-dev/neo-go@v0.121.0/pkg/compiler/compiler.go:130
github.com/nspcc-dev/neo-go/pkg/compiler.(*codegen).compile(0x1d381ae145a0, 0x1d381ad2db60, 0x1?)
        github.com/nspcc-dev/neo-go@v0.121.0/pkg/compiler/codegen.go:2496 +0x865
github.com/nspcc-dev/neo-go/pkg/compiler.codeGen(0x1d381ad2db60)
        github.com/nspcc-dev/neo-go@v0.121.0/pkg/compiler/codegen.go:2550 +0x379
github.com/nspcc-dev/neo-go/pkg/compiler.CompileWithOptions({0x7ffe0eba8ba5?, 0x3?}, {0x0?, 0x0?}, 0x1d381a9ea000)
        github.com/nspcc-dev/neo-go@v0.121.0/pkg/compiler/compiler.go:249 +0x67
github.com/nspcc-dev/neo-go/pkg/compiler.CompileAndSave({0x7ffe0eba8ba5, 0xf}, 0x1d381a9ea000)
        github.com/nspcc-dev/neo-go@v0.121.0/pkg/compiler/compiler.go:265 +0x20b
github.com/nspcc-dev/neo-go/cli/smartcontract.contractCompile(0x1d381a9be580)
        github.com/nspcc-dev/neo-go@v0.121.0/cli/smartcontract/smart_contract.go:545 +0xac5
github.com/urfave/cli/v2.(*Command).Run(0x1d381a9bc9a0, 0x1d381a9be580, {0x1d381a9bae70, 0xb, 0xb})
        github.com/urfave/cli/v2@v2.27.7/command.go:276 +0x7c2
github.com/urfave/cli/v2.(*Command).Run(0x1d381a9bc840, 0x1d381a9be480, {0x1d381a9cc6c0, 0xc, 0xc})
        github.com/urfave/cli/v2@v2.27.7/command.go:269 +0xa30
github.com/urfave/cli/v2.(*Command).Run(0x1d381a9d2dc0, 0x1d381a9be340, {0x1d381a878340, 0xd, 0xd})
        github.com/urfave/cli/v2@v2.27.7/command.go:269 +0xa30
github.com/urfave/cli/v2.(*App).RunContext(0x1d381a9ae000, {0x1510950, 0x20b5b60}, {0x1d381a878340, 0xd, 0xd})
        github.com/urfave/cli/v2@v2.27.7/app.go:333 +0x5a5
github.com/urfave/cli/v2.(*App).Run(0x0?, {0x1d381a878340?, 0x70?, 0x0?})
        github.com/urfave/cli/v2@v2.27.7/app.go:307 +0x2f
main.main()
        github.com/nspcc-dev/neo-go@v0.121.0/cli/main.go:13 +0x32
make: *** [Makefile:35: contracts/neofs/contract.nef] Ошибка 2
```

## Expected Behavior
No panic.

## Possible Solution
Fix something.

## Steps to Reproduce
Update contracts to 0.121.0, compile neofs (PR to be linked).

## Context
Breaks neofs-contract.

## Regression
Yes, worked with 0.117.0.

## Your Environment
* Version of the product used: 0.121.0

--- a/pkg/compiler/compiler.go
+++ b/pkg/compiler/compiler.go
@@ -216,10 +216,23 @@ func getBuildInfo(name string, src any) (*buildInfo, error) {
 	if err != nil {
 		return nil, err
 	}
-	for _, p := range prog {
-		if len(p.Errors) != 0 {
-			return nil, p.Errors[0]
+	// packages.Load never returns per-package errors via err: they sit in each
+	// Package's own Errors field, including transitive imports, so the whole
+	// graph must be walked to find them. Check in pre, not post, to stop
+	// descending as soon as an erroneous package is found.
+	var firstErr error
+	packages.Visit(prog, func(p *packages.Package) bool {
+		if firstErr != nil {
+			return false
 		}
+		if len(p.Errors) != 0 {
+			firstErr = p.Errors[0]
+			return false
+		}
+		return true
+	}, nil)
+	if firstErr != nil {
+		return nil, firstErr
 	}
 	return &buildInfo{
 		config:  conf,

--- a/pkg/compiler/import_test.go
+++ b/pkg/compiler/import_test.go
@@ -89,3 +89,13 @@ func TestImportCycleIndirect(t *testing.T) {
 	_, _, err := compiler.CompileWithOptions("some.go", strings.NewReader(src), nil)
 	require.Error(t, err)
 }
+
+func TestImportPackageError(t *testing.T) {
+	src := `package foo
+		import "github.com/nspcc-dev/neo-go/pkg/compiler/testdata/importerr"
+		func Main() []importerr.Ballot {
+			return importerr.GetBallots()
+		}`
+	_, err := compiler.Compile("foo.go", strings.NewReader(src))
+	require.ErrorContains(t, err, "data (variable of type []byte) is not an interface")
+}

--- a/pkg/compiler/testdata/importerr/importerr.go
+++ b/pkg/compiler/testdata/importerr/importerr.go
@@ -1,0 +1,22 @@
+package importerr
+
+import (
+	"github.com/nspcc-dev/neo-go/pkg/interop/native/std"
+	"github.com/nspcc-dev/neo-go/pkg/interop/storage"
+)
+
+type Ballot struct {
+	X int
+}
+
+// GetBallots contains an invalid type assertion: storage.LocalGet already
+// returns []byte, so data.([]byte) doesn't type-check (data is not an
+// interface). This must be reported as a normal compile error, not silently
+// ignored because the error is in an imported package.
+func GetBallots() []Ballot {
+	data := storage.LocalGet([]byte("votekey"))
+	if data != nil {
+		return std.Deserialize(data.([]byte)).([]Ballot)
+	}
+	return []Ballot{}
+}

